src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -174,7 +174,8 @@ internal bool MustHaveRequestBody
                 // Normalize before calling this
                 Debug.Assert(ReferenceEquals(this, Normalize(this)));
 
-                return !ReferenceEquals(this, HttpMethod.Get) && !ReferenceEquals(this, HttpMethod.Head) && !ReferenceEquals(this, HttpMethod.Connect);
+                return !ReferenceEquals(this, HttpMethod.Get) && !ReferenceEquals(this, HttpMethod.Head) && !ReferenceEquals(this, HttpMethod.Connect) &&
```

Comment:
Not a change request just an observation, you could use `is` here to make this cleaner.
`return !(this is HttpMethod.Get || this is HttpMethod.Head || this is HttpMethod.Connect || this is HttpMethod.Options || this is HttpMethod.Delete); `

Reply:
Why ReferenceEquals? Why not String.Equals()? Does this work with
`var method = new System.Net.Http.HttpMethod("GET");` ?